src/core/arm/arm_interface.h

```diff
@@ -19,6 +24,46 @@ class ARM_Interface : NonCopyable {
     virtual ~ARM_Interface() {}
 
     class ThreadContext {
+        friend class boost::serialization::access;
+
+        template <class Archive>
+        void save(Archive& ar, const unsigned int file_version) const {
+            for (auto i = 0; i < 16; i++) {
+                auto r = GetCpuRegister(i);
+                ar << r;
+            }
+            for (auto i = 0; i < 16; i++) {
+                auto r = GetFpuRegister(i);
```

Comment:
```suggestion
                const auto r = GetFpuRegister(i);
```

etc where applicable below.